### PR TITLE
Fixes runechat layering issues

### LINKED ISSUE #26861: Runechat doesn't work when the atom speaks from a non-turf loc
(WEB REPORT BY: shiftyrail REMOTE: 206.221.180.138:7777)
# Revision
9ea0c1a308b7e694f9771fc52ee0dddc9f799e8e

# Description

For instance, if you are holding a recorder, the images will be generated at the bottom of your screen, at the level of the hands HuD.

This could be a neat feature for audio-logs or something like that, but the other players should have it displayed on their screen if they can see the object.

### LINKED ISSUE #26867: Runechat is displayed under some items, blocking readability
From what i've seen curtains, security cameras, and houseplants.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Tentatively closes #26861. 

pAIs speaking to their masters also enjoy some privacy. I think the effect looks good. The message will not appear in the game world.

Closes #26867 

runechat_in_hand

### CODE REVIEW

File: code/datums/chat_message.dm [hunk 1]
```diff
@@ -138,6 +138,9 @@
 	message.maptext_x = (CHAT_MESSAGE_WIDTH - owner.bound_width) * -0.5
 	message.maptext = complete_text
 
+	if (target.loc == owner) // Special case, holding an atom speaking (pAI, recorder...)
```
Comment (DamianX) [inline code/datums/chat_message.dm]: This looks like it wouldn't handle cases where the item is nested more than 1 level deep, perhaps `is_holder_of` is better?

Comment (ShiftyRail) [reply code/datums/chat_message.dm]: Correct, however it wouldn't show up at all as the screen_loc of the atom would be in nullspace. The pAI speaking in the backpack would be hidden in the same way that someone speaking in a closed locker is hidden.

Comment (DamianX) [reply code/datums/chat_message.dm]: What about a pAI which is in a box which is in a backpack?

Comment (ShiftyRail) [reply code/datums/chat_message.dm]: Did that + the optimisations you suggested (notably, caching the icon)

File: code/datums/chat_message.dm [hunk 2]
```diff
@@ -138,6 +144,9 @@
 	message.maptext_x = (CHAT_MESSAGE_WIDTH - owner.bound_width) * -0.5
 	message.maptext = complete_text
 
+	if (is_holder_of(owner, target)) // Special case, holding an atom speaking (pAI, recorder...)
+		message.plane = ABOVE_HUD_PLANE
+
 	// View the message
 	owned_by.seen_messages.Add(src)
 	owned_by.images |= message
```
Comment (DamianX) [inline code/datums/chat_message.dm]:
```suggestion
	owned_by.images += message
```

**Pull request merged**